internal/logger/logger.go

```diff
@@ -0,0 +1,106 @@
+package logger
+
+import (
+	"io"
+	"log/slog"
+	"strings"
+)
+
+// Service holds the logger and its dynamic level controller.
+type Service struct {
+	*slog.Logger
+	level *slog.LevelVar
+}
+
+// SetLevel dynamically changes the logging level.
+func (s *Service) SetLevel(level string) {
+	s.level.Set(parseLevel(level))
+}
+
+// New creates a new logging service.
+func New(level, format string, writer io.Writer) *Service {
+	levelVar := &slog.LevelVar{}
+	levelVar.Set(parseLevel(level))
+
+	opts := &slog.HandlerOptions{
+		Level:       levelVar,
+		ReplaceAttr: replaceAttr,
+	}
+
+	var handler slog.Handler
+	switch strings.ToLower(format) {
+	case "json":
+		handler = slog.NewJSONHandler(writer, opts)
+	default:
+		handler = slog.NewTextHandler(writer, opts)
```

Comment:
I would handle the log format/level explicitly and logging the what the user passed as environment variable, right now, if you pass an invalid "LOG_FORMAT" or "LOG_LEVEL" the user has no visibility of the mistake!